### PR TITLE
feat: Implement email sending functionalities

### PR DESCRIPTION
**Pull request by King-Mikaelson**
## What does the PR do?
This PR is about creating an email service for sending various types of email
## How should this be manually tested?
**call the email service with expected parameters email, url, token and articles**
```
sendUserConfirmation(email: string, url: string, token: string)

sendForgotPassword(email: string, url: string, token: string)

sendWaitListMail(email: string, url: string)

sendNewsLetterMail(email: string, articles: Article[])
```

## Checklist of what you did
- [x] Implemented a function  to confirm user registration via mail
- [x]  Implemented a function  to send forget password mail 
- [x] Implemented a function  to send waitlist email
- [x] Implemented a function  to send for newsletters
- [x] Added unit tests for email service

## Reference Issue
[Reset Password and Forget Password](https://github.com/hngprojects/hng_boilerplate_nestjs/issues/12)

### CODE REVIEW

**Comment by davidolanrewaju** [comment]
Please follow the PR pattern for #351, and also include documentations

**Comment by King-Mikaelson** [comment]
> Please follow the PR pattern for #351, and also include documentations

Hi this is just a service there is no endpoint,other endpoints depend on it

**Review by Heba-WebDev — APPROVED** [review]
Well done, Michael. Just please update the title of the issue and your description to match the format.

**Comment by King-Mikaelson** [comment]
> Well done, Michael. Just please update the title of the issue and your description to match the format.

Done thank you.Kindly review again

**Comment by King-Mikaelson** [comment]
> Please follow the PR pattern for #351, and also include documentations

I have made changes kindly review again

**Comment by Heba-WebDev** [comment]
@King-Mikaelson resolve the merge conflicts please

**Review by davidolanrewaju — APPROVED** [review]

**Comment by King-Mikaelson** [comment]
> @King-Mikaelson resolve the merge conflicts please

All merge conflicts has been resolved please review thank you

File: src/modules/email/article.interface.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export interface Article {
```
Comment (Homoakin619) [inline src/modules/email/article.interface.ts:1]: rename to ArticleInterface

File: src/modules/email/email.controller.ts
Comment (Homoakin619) [inline src/modules/email/email.controller.ts:1]: This controller is redundant, kindly remove

File: src/modules/email/email.service.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
+// src/email/email.service.ts
+import { Injectable } from '@nestjs/common';
+import { MailerService } from '@nestjs-modules/mailer';
+import { Article } from './article.interface';
+
+@Injectable()
+export class EmailService {
+  constructor(private readonly mailerService: MailerService) {}
+
+  async sendUserConfirmation(email: string, url: string, token: string) {
```
Comment (Homoakin619) [inline src/modules/email/email.service.ts:10]: Make this more descriptive like sendUserConfirmationMail, and do the same to others as well

**Review by Homoakin619 — COMMENTED** [review]
@buka4rill kindly review

**Comment by King-Mikaelson** [comment]
> @buka4rill I have reviewed, awaiting your approval

@Homoakin619 please approve the changes too thank you

File: src/modules/email/email.module.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+// src/email/email.module.ts
```
Comment (buka4rill) [inline src/modules/email/email.module.ts:1]: Remove this

File: src/modules/email/email.service.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
+// src/email/email.service.ts
```
Comment (buka4rill) [inline src/modules/email/email.service.ts:1]: Remove this

**Comment by Homoakin619** [comment]
@King-Mikaelson your code is failing pipeline tests

**Review by buka4rill — COMMENTED** [review]
Small comments

**Comment by King-Mikaelson** [comment]
> Small comments

Thank you buka resolved all issues plus test cases

**Comment by King-Mikaelson** [comment]
> @King-Mikaelson your code is failing pipeline tests

resolved kindly check again

**Review by buka4rill — APPROVED** [review]
LGTM ✨

**Pull request merged**